gatsby-node.js

```diff
@@ -1,27 +1,17 @@
+require('dotenv').config()
 const {
-  getNodeSlug,
   setPageContext,
   removePageTrailingSlash
 } = require('./src/gatsby/common')
-const { createPages: createDocPages } = require('./src/gatsby/doc')
-const { createPages: createBlogPages } = require('./src/gatsby/blog')
 
-exports.onCreateNode = ({ node, actions, getNode }) => {
-  const { createNodeField } = actions
+const models = require('./src/gatsby/models.js')
+const runOnModels = require('./src/gatsby/utils/models')
 
-  if (node.internal.type === 'MarkdownRemark') {
-    createNodeField({
-      name: 'slug',
-      node,
-      value: getNodeSlug(node, getNode)
-    })
-  }
-}
-
-exports.createPages = async ({ graphql, actions }) => {
-  createDocPages({ graphql, actions })
-  createBlogPages({ graphql, actions })
-}
+exports.createSchemaCustomization = api =>
+  runOnModels(models, 'createSchemaCustomization', api)
+exports.onCreateNode = api =>
+  runOnModels(models, 'onCreateNode', api, { models })
```

Comment:
do we need the last `{ models }`?

Reply:
We do need it, because Models that run other Models cannot import the site's Models from the module and run them. I tried switching the dependency injection with a require for aesthetics, but it ends up leading to a circular dependency and breaking the build. It took me a while to figure out what was going on.

Reply:
We don't, I had it like this before I added the bit that passes it by default. Good catch!